app/src/main/java/dev/msfjarvis/aps/ui/crypto/DecryptActivity.kt

```diff
@@ -165,55 +165,54 @@ class DecryptActivity : BasePgpActivity(), OpenPgpServiceConnection.OnBound {
     val inputStream = File(fullPath).inputStream()
     val outputStream = ByteArrayOutputStream()
 
-    lifecycleScope.launch(Dispatchers.IO) {
-      api?.executeApiAsync(data, inputStream, outputStream) { result ->
-        when (result?.getIntExtra(OpenPgpApi.RESULT_CODE, OpenPgpApi.RESULT_CODE_ERROR)) {
-          OpenPgpApi.RESULT_CODE_SUCCESS -> {
-            startAutoDismissTimer()
-            runCatching {
-              val showPassword = settings.getBoolean(PreferenceKeys.SHOW_PASSWORD, true)
-              val entry = passwordEntryFactory.create(lifecycleScope, outputStream.toByteArray())
-              val items = arrayListOf<FieldItem>()
-              val adapter = FieldItemAdapter(emptyList(), showPassword) { text -> copyTextToClipboard(text) }
-
-              if (settings.getBoolean(PreferenceKeys.COPY_ON_DECRYPT, false)) {
-                copyPasswordToClipboard(entry.password)
-              }
+    lifecycleScope.launch(Dispatchers.Main) {
+      val result = withContext(Dispatchers.IO) { checkNotNull(api).executeApi(data, inputStream, outputStream) }
+      when (result.getIntExtra(OpenPgpApi.RESULT_CODE, OpenPgpApi.RESULT_CODE_ERROR)) {
+        OpenPgpApi.RESULT_CODE_SUCCESS -> {
+          startAutoDismissTimer()
+          runCatching {
+            val showPassword = settings.getBoolean(PreferenceKeys.SHOW_PASSWORD, true)
+            val entry = passwordEntryFactory.create(lifecycleScope, outputStream.toByteArray())
+            val items = arrayListOf<FieldItem>()
+            val adapter = FieldItemAdapter(emptyList(), showPassword) { text -> copyTextToClipboard(text) }
+
+            if (settings.getBoolean(PreferenceKeys.COPY_ON_DECRYPT, false)) {
+              copyPasswordToClipboard(entry.password)
+            }
 
-              passwordEntry = entry
-              invalidateOptionsMenu()
+            passwordEntry = entry
+            invalidateOptionsMenu()
 
-              if (!entry.password.isNullOrBlank()) {
-                items.add(FieldItem.createPasswordField(entry.password!!))
-              }
+            if (!entry.password.isNullOrBlank()) {
+              items.add(FieldItem.createPasswordField(entry.password!!))
+            }
 
-              if (entry.hasTotp()) {
-                launch(Dispatchers.IO) {
-                  withContext(Dispatchers.Main) {
-                    val code = entry.totp.value
-                    items.add(FieldItem.createOtpField(code))
-                  }
-                  entry.totp.collect { code -> withContext(Dispatchers.Main) { adapter.updateOTPCode(code) } }
+            if (entry.hasTotp()) {
+              launch(Dispatchers.IO) {
+                withContext(Dispatchers.Main) {
+                  val code = entry.totp.value
+                  items.add(FieldItem.createOtpField(code))
                 }
+                entry.totp.collect { code -> withContext(Dispatchers.Main) { adapter.updateOTPCode(code) } }
               }
```

Comment:
```suggestion
              launch(Dispatchers.Main) {
                val code = entry.totp.value
                items.add(FieldItem.createOtpField(code))
                entry.totp.collect { code -> adapter.updateOTPCode(code) }
            }
```

I believe we can just launch the coroutine in `Dispatchers.Main` since all the work is already happening the `Main` context.

Reply:
Yeah this is a holdover from the switch I made to the outermost scope.